visualization/dashboard/src/MLIDashboard/Controls/WhatIfTab/WhatIfTab.tsx

```diff
@@ -246,6 +247,8 @@ export class WhatIfTab extends React.PureComponent<IWhatIfTabProps, IWhatIfTabSt
             this.props.chartProps,
             this.props.cohorts[this.state.selectedCohortIndex]
         );
+        const cohortLength = this.props.cohorts[this.state.selectedCohortIndex].rowCount;
+        const canRenderChart = cohortLength < NewExplanationDashboard.ROW_ERROR_SIZE || this.props.chartProps.chartType !== ChartTypes.Scatter;
```

Comment:
see some common properties/code shared by these tabs, can these be moved to a more common place to limit duplicate code?

Reply:
I don't feel the current checks and presentation need to be extracted more than they are (strings and styles are shared) since the decision on when to show a warning rather than th chart will be view specific. The warning itself is not, so that is all shared.